extensions/sql-migration/src/scripts/SHIR-auto-configuration.ps1

```diff
@@ -0,0 +1,609 @@
+﻿###############################################################################################
+# $Description: This script automatically downloads, installs and configures SHIR to DMS on a Windows machine.
+# $Id: SHIR-auto-configuration.ps1
+# $Author: anjaligoyal $ gasachdeva $
+###############################################################################################
+
+<#
+.SYNOPSIS
+Automatically configures SHIR on a Windows machine including downloading, installing Integration runtime,
+and configuring IR with DMS.
+
+.DESCRIPTION
+The script downloads, installs and configures SHIR on a Windows machine.
+
+.PARAMETER AdminPriv
+(Optional) [switch] True if script is running with admin privileges else false.
+
+.PARAMETER AuthKey1
+(Optional) [string] First authentication key to register node to DMS.
+
+.PARAMETER AuthKey2
+(Optional) [string] Second authentication key to register node to DMS.
+#>
+
+param (
+	[switch] $AdminPriv = $false,
+	[string] $AuthKey1 = $null,
+	[string] $AuthKey2 = $null
+)
+
+$timeStamp = [System.DateTime]::Now.ToString("yyyyMMddHHmmss")
+# Unique script id for every run - Telemetry purpose
+$Global:ScriptId = "Script-$timestamp"
+
+# TODO: Auto populate
+$Global:LatestIRVersion = [Version]"5.35.8686.1"
+
+# Minimum NuGet version required to install Az.DataMigration module
+$Global:MinRequiredNuGetVersion = [Version]"2.8.5.201"
+
+# Log file path
+$Global:Logfile = [System.IO.Path]::Combine($env:USERPROFILE, "shir", "shir-$timeStamp.log")
+
+# Guid used to create a script lock (to disable concurrent script instances)
+$Global:InstallationLockGUID = $MyInvocation.MyCommand.Name
+
+################################### Main Script Block ##################################
+$main = {
+	$isFirstInstance = $false
+	# Create a new Stopwatch object
+	$stopwatch = New-Object System.Diagnostics.Stopwatch
+	# Start the stopwatch
+	$stopwatch.Start()
+
+	try {
+		New-Item -ItemType File -Path $Global:Logfile -Force | Out-Null
+		# Create a lock file so that only one instance of script can run at a time
+		$isFirstInstance = Lock-ScriptInstance
+		if ($isFirstInstance) {
+			Install-IR
+		}
+	}
+	catch {
+		Write-ErrorAndLog -exception $_.Exception
+		Write-ErrorAndLog "The script could not configure SHIR due to an internal exception. Please try again."
+	}
+	finally {
+		# Only allow the first running instance to unlock the script
+		if ($isFirstInstance) {
+			Unlock-ScriptInstance
+		}
+		# Stop the stopwatch
+		$stopwatch.Stop()
+
+		# Write the total time taken
+		Write-OutputAndLog ("Total time taken: {0}" -f $stopwatch.Elapsed)
```

Comment:
please add unit for the time (minuets, milliseconds, seconds etc.)

Reply:
![image](https://github.com/microsoft/azuredatastudio/assets/66509246/9dfdcf80-09a9-4e12-9570-f387c118604d)

Reply:
This looks good then. probably add (hh:mm:ss) as the format in the log

Reply:
Done
<img width="857" alt="image (2)" src="https://github.com/microsoft/azuredatastudio/assets/66509246/b000d707-aa24-4cd2-9732-92f73ddfa4ab">